server-spi-private/src/test/java/org/keycloak/models/MigrationVersionTest.java

```diff
@@ -82,6 +81,17 @@ public void testVersion() {
 
         Assert.assertTrue(version_211CR1.lessThan(version50Snapshot));
 
+        ModelVersion versionPipeline = new ModelVersion("8.0.2-REL-20200130-143126");
+        Assert.assertEquals(8, versionPipeline.getMajor());
+        Assert.assertEquals(0, versionPipeline.getMinor());
+        Assert.assertEquals(2, versionPipeline.getMicro());
+        Assert.assertNull(versionPipeline.getQualifier());
+
+        ModelVersion versionProduct = new ModelVersion("7.0.0.redhat-00002");
+        Assert.assertEquals(7, versionProduct.getMajor());
+        Assert.assertEquals(0, versionProduct.getMinor());
+        Assert.assertEquals(0, versionProduct.getMicro());
+        Assert.assertNull(versionProduct.getQualifier());
```

Comment:
NIT: There is another format of version what we're using in pipeline: `9.0.0-SNAPSHOT-stage-20191125-003440` . It might be worth to test it as well. But it can go in as it is.

Reply:
Added that one as well